### PR TITLE
Product Query - Add E2E tests for the Filter by Attribute  block

### PR DESCRIPTION
**Pull request by gigitux**
This PR adds e2e tests for the Filter By Attribute Block and Product Query. 

Blocked by #7351 

#### Automated Tests
* [x] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

#### User Facing Testing

1. Be sure that the E2E tests don't fail.

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental

### CODE REVIEW

File: tests/e2e/specs/shopper/filter-products-by-attribute.test.ts [hunk 1]
```diff
@@ -227,4 +229,121 @@ describe( `${ block.name } Block`, () => {
 			);
 		} );
 	} );
+
+	/*
+	 * @todo Remove this logic when WordPress 6.1 is released. This is a temporary fix because WP 6.0 doesn't ship the necessary changes to make it work properly the Product Query block.
+	 */
```
Comment (Aljullu) [inline tests/e2e/specs/shopper/filter-products-by-attribute.test.ts]: I'm not sure to understand this comment, does it refer to the entire `describe()` clause? If so, why it wasn't necessary in the _Filter by Price_ block tests?

File: tests/e2e/specs/shopper/filter-products-by-attribute.test.ts [hunk 2]
```diff
@@ -227,4 +229,121 @@ describe( `${ block.name } Block`, () => {
...
+
+			expect( isRefreshed ).toBeCalledTimes( 1 );
+			expect( products ).toHaveLength( 1 );
+			await expect( page ).toMatch( block.foundProduct );
+			expect( parsedURL.search ).toEqual(
+				block.urlSearchParamWhenFilterIsApplied
+			);
+		} );
+
+		it( 'should refresh the page only if the user click on button', async () => {
```
Comment (Aljullu) [inline tests/e2e/specs/shopper/filter-products-by-attribute.test.ts]:
```suggestion
		it( 'should refresh the page only if the user clicks on button', async () => {
```

File: tests/e2e/specs/shopper/filter-products-by-attribute.test.ts [hunk 3]
```diff
@@ -227,4 +229,121 @@ describe( `${ block.name } Block`, () => {
...
+			await publishPost();
+
+			editorPageUrl = page.url();
+			frontedPageUrl = await page.evaluate( () =>
+				wp.data.select( 'core/editor' ).getPermalink()
+			);
+			await page.goto( frontedPageUrl );
+		} );
+
+		it( 'should render', async () => {
```
Comment (Aljullu) [inline tests/e2e/specs/shopper/filter-products-by-attribute.test.ts]: Same as in the other PR, should we rename this to `should render products`?

**Review by Aljullu — APPROVED** [review]
I rebased this PR with the latest trunk to fix the conflicts and make it easier for me to review. Hope I didn't mess anything. :slightly_smiling_face: 

Tests are passing and code looks good, I left a couple of questions (some similar to the Filter by Price PR https://github.com/woocommerce/woocommerce-blocks/pull/7351), but pre-approving anyway.

File: tests/e2e/specs/shopper/filter-products-by-attribute.test.ts (hunk 1 above)
Comment (gigitux) [reply tests/e2e/specs/shopper/filter-products-by-attribute.test.ts]: I forgot to remove this comment! I worked on this PR before that WordPress 6.1 was released. Now, it isn't necessary anymore!

**Comment by gigitux** [comment]
Thanks a lot for the review and for taking care to resolve the conflicts!
I addressed your feedback! I will merge the PR! 💪

**Pull request merged**